service/controller/v9patch1/adapter/auto_scaling_group.go

```diff
@@ -27,7 +27,7 @@ func (a *autoScalingGroupAdapter) getAutoScalingGroup(cfg Config) error {
 	}
 
 	a.WorkerAZ = key.AvailabilityZone(cfg.CustomObject)
-	a.ASGMaxSize = workers
+	a.ASGMaxSize = workers + 1
```

Comment:
Shouldn't that be new patch?

Reply:
This is backported to make it work at all. So yeah we should actually create a new patch but then we cannot upgrade to it because the upgrade is broken.

Reply:
I have upgraded one of the asgard clusters to v9patch1. Wondering what will happen with existing cluster? Will it reconcile and (hopefully) no vms will be rolled out.

I'm ok with this change in `v9patch1` at that point in time, taking into account one more adjustment needed.

https://github.com/giantswarm/giantswarm/issues/3160

One thing that we need to test this beforehand.

Reply:
This is only relevant for clusters having only one worker. I only know of 1 customer guest cluster with this special condition. Then the upgrade depends on the version bundle version, not on such a single property. See https://github.com/giantswarm/aws-operator/blob/785b310d8c9c60aa3e4c92b705abef215bcc2c11/service/controller/v11/resource/cloudformation/update.go#L224-L243.